compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -145,7 +144,8 @@ mod opaque_types;
 mod parameterized;
 mod rvalue_scopes;
 mod structural_impls;
-mod sty;
+#[path = "sty.rs"]
+mod sty_;
```

Comment:
@Nilstrieb suggested allowing the lint in this case and a couple of other cases https://github.com/rust-lang/rust/pull/111378/files#r1205160776

Reply:
Right, I will change it back and just allow the lint